### ISSUE TITLE
ExecutionFailed error when renamed identifier is used in specific MATCH and WHERE statements

### ISSUE THREAD

**Issue by slbelden** [neo4j/neo4j#9784]
Attempting to execute the following Cypher text yields a Neo.DatabaseError.Statement.ExecutionFailed error:

```
MATCH (node:Type)
WITH node AS point
MATCH (parent:ParentType)
MATCH (point)-[:MEMBER_OF]->(group:GroupType)
WHERE (point)-[:CHILD_OF]->(parent)
RETURN point
```

The error occurs regardless of any data in the database or the existence of any of the labels used.

**Neo4j Version:** 3.2.1, 3.2.3
**Operating System:** Windows 10
**API:** N/A

### Steps to reproduce

1. Run the given Cypher query against any Neo4j database.
    * The specified labels and relationships do not have to exist. In fact nothing has to exist, the error occurs when run against an empty database exactly the same as a database containing all the expected data.

### Expected behavior

With valid data, `point` is returned.

With invalid or no data, the query successfully executes but nothing is returned.

Examples of Cypher queries with identical behavior that do work as expected:

```
MATCH (point:Type)
MATCH (parent:ParentType)
MATCH (point)-[:MEMBER_OF]->(group:GroupType)
WHERE (point)-[:CHILD_OF]->(parent)
RETURN point
```

(`WITH` rename removed)

```
MATCH (node:Type)
WITH node AS point
MATCH (point)-[:MEMBER_OF]->(group:GroupType)
MATCH (point)-[:CHILD_OF]->(parent:ParentType)
RETURN point
```

(`MATCH` of `parent` inlined)

```
MATCH (node:Type)
WITH node AS point
MATCH (parent:ParentType)
MATCH (group:GroupType)
WHERE (point)-[:MEMBER_OF]->(group)
AND (point)-[:CHILD_OF]->(parent)
RETURN point
```

(`MATCH` of `group` moved out of relationship match)

### Actual behavior

```
ERROR: Neo.DatabaseError.Statement.ExecutionFailed

Expected 
RegularPlannerQuery(QueryGraph(Set(),Set(IdName(node)),Set(),Selections(Set(Predicate(Set(IdName(node)),HasLabels(Variable(node),List(LabelName(Type)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(point -> Variable(node)),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(  UNNAMED77),(IdName(point),IdName(group)),OUTGOING,List(RelTypeName(MEMBER_OF)),SimplePatternLength)),Set(IdName(parent), IdName(point), IdName(group)),Set(IdName(point)),Selections(Set(Predicate(Set(IdName(point), IdName(parent)),PatternExpression(RelationshipsPattern(RelationshipChain(NodePattern(Some(Variable(point)),List(),None),RelationshipPattern(None,List(RelTypeName(CHILD_OF)),None,None,OUTGOING,false),NodePattern(Some(Variable(parent)),List(),None))))), Predicate(Set(IdName(group)),HasLabels(Variable(group),List(LabelName(GroupType)))), Predicate(Set(IdName(parent)),HasLabels(Variable(parent),List(LabelName(ParentType)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(point -> Variable(point)),QueryShuffle(List(),None,None)),None))) 


Instead, got: 
RegularPlannerQuery(QueryGraph(Set(),Set(IdName(node)),Set(),Selections(Set(Predicate(Set(IdName(node)),HasLabels(Variable(node),List(LabelName(Type)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(point -> Variable(node)),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(  UNNAMED77),(IdName(point),IdName(group)),OUTGOING,List(RelTypeName(MEMBER_OF)),SimplePatternLength)),Set(IdName(point), IdName(group)),Set(IdName(point)),Selections(Set(Predicate(Set(IdName(group)),HasLabels(Variable(group),List(LabelName(GroupType)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(point -> Variable(point)),QueryShuffle(List(),None,None)),None)))
```

**Comment by jexp** [comment]
Could you try to reproduce it with the latest version?
 e.g. 3.0.11 and 3.2.3

Thanks

**Comment by slbelden** [comment]
Neo4j version was incorrectly reported as the Neo4j Browser version in the original issue. Issue actually occurred in 3.2.1. Upgrading to 3.2.3 yields identical issue.

**Comment by sherfert** [comment]
Hi. This has been fixed in 3.2.6.
